fix oai ia imports

summer-cook opened the pull request:
# Coauthor
@aprilrieger

# Summary
imports with the oai ia parser were failing. This PR upgrades some of the overridden bulkrax files to fix these imports. 

# Related
https://github.com/scientist-softserv/atla_digital_library/issues/248

# Screenshot
can now import successfully with the OaiIA parser: 
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/73361970/233202067-149fee96-7557-4b91-89ef-b8879a49b090.png">


# Acceptance Criteria
- [ ] as a user, I can import an oai feed with the oai ia parser

# Testing Instructions
1. Sign in and go to the /importers page
2. Click the button to create a new importer
3. Use the following applicable info to fill out the importer:

name: `AMBS -  Biblioteca Digital Anabautista`
admin_set_id: `admin_set/default`
limit: `25` <---- set this or you will get ALOT of entries
parser: `OAI - Internet Archive`
base_url: `https://archive.org/services/oai.php`
metadata_prefix: `oai_dc`
set: `collection:bibliotecadigitalanabautista`
collection_title: `AMBS -  Biblioteca Digital Anabautista`
institution_name: `Anabaptist Mennonite Biblical Seminary`
thumbnail_url: `https://archive.org/services/img/<%= identifier.split(':').last %>"}`

4. Ensure the importer says "completed" and works import successfully, report any errors you see with screenshots and behavior description, try to include as much detail as possible so we can pickup where you left off if it gets kicked back :)

summer-cook committed:
replace system_identifier_field with work_identifier, change ImportWorkCollectionJob to ImportCollectionJob, upgrades to create_collections in parsers

summer-cook committed:
add decorator for application parser for patch

aprilrieger committed:
Updates the class to module for decorator pattern

aprilrieger committed:
Appease the coppahs

aprilrieger committed:
Remove comments

bkiahstroud commented on app/parsers/bulkrax/application_parser_decorator.rb:
```diff
@@ -0,0 +1,16 @@
+    module ApplicationParserDecorator
+      # override create_objects to get the patch from https://github.com/samvera-labs/bulkrax/pull/663
+      def create_objects(types = [])
+        types.each do |object_type|
```
1. Without the `DEFAULT_OBJECT_TYPES` variable added by https://github.com/samvera-labs/bulkrax/pull/663/files, what does `types` end up being here? 
2. Also, can we please add a `TODO` comment that says to delete this decorator when upgrading Bulkrax to v5.x.x?

summer-cook replied:
I realized i just got the PR in the comment wrong. PR #663 IS already in version 4.4.1, its this patch PR https://github.com/samvera-labs/bulkrax/pull/713 that we needed to pull in.

summer-cook replied:
updating comments now though to also add the todo and fix the PR that is linked there!

summer-cook replied:
@bkiahstroud done!!

summer-cook committed:
fix comments in decorator

summer-cook committed:
fix comments in decorator

summer-cook committed:
update comment again

summer-cook committed:
decorator should be a module, not a class. dc up would not start with it as a class

summer-cook committed:
remove controller since the overrides are no longer needed. we will now inherit directly from bulkrax

summer-cook commented:
@bkiahstroud i updated this branch with your comments and also realized i forgot to remove the overridden controller, so i added another commit.

reason for removal: all the overrides were no longer needed & had been either updated or added in in the version of bulkrax we upgraded to. so its better now to just have that file inherit.

summer-cook committed:
Merge branch 'main' of github.com:scientist-softserv/atla_digitial_library into 248-oai-la-parser-issues

summer-cook committed:
Merge branch 'main' of github.com:scientist-softserv/atla_digitial_library into 248-oai-la-parser-issues

bkiahstroud commented:
@summer-cook @aprilrieger I see this was recently marked as a draft, is it ready for re-review?

aprilrieger committed:
Updates with a good oaiia_importer

summer-cook committed:
Merge branch '248-oai-la-parser-issues' of github.com:scientist-softserv/atla_digitial_library into 248-oai-la-parser-issues

summer-cook commented:
> @summer-cook @aprilrieger I see this was recently marked as a draft, is it ready for re-review?

not yet, just trying to figure out a few more things

aprilrieger committed:
Merge branch 'main' into 248-oai-la-parser-issues

aprilrieger committed:
default create_relationships & create_file_sets, removed the importer controller, removed the decorator, removed the overridden partial as unecessary now

summer-cook committed:
Merge branch '248-oai-la-parser-issues' of github.com:scientist-softserv/atla_digitial_library into 248-oai-la-parser-issues

alishaevn commented on config/initializers/bulkrax.rb:
```diff
@@ -14,7 +14,7 @@
 
   # Field to use during import to identify if the Work or Collection already exists.
-  # Default is 'source'.
+  # Default is 'work_identifier' which is 'source'
```
```suggestion
  # Default is 'source'; defined in #work_identifier.
```

summer-cook committed:
updates bulkrax to 4.4.2 to get changes

summer-cook committed:
Update config/initializers/bulkrax.rb

Co-authored-by: Alisha Evans <alisha@scientist.com>

summer-cook committed:
make rubocop happy

summer-cook committed:
Merge branch '248-oai-la-parser-issues' of github.com:scientist-softserv/atla_digitial_library into 248-oai-la-parser-issues

summer-cook commented:
@bkiahstroud ok, NOW it is ready

bkiahstroud commented on app/parsers/bulkrax/oai_ia_parser.rb:
```diff
@@ -1,3 +1,6 @@
+# OVERRIDE: Bulkrax 4.4.1 to add collection title, noted below.
+# OVERRIDE: Bulkrax 4.4.1 to get a patch from Bulkrax 5.x.x for create_objects, noted below.
+# TODO: remove create_objects method when upgrading to Bulkrax 5.x.x
```
@summer-cook these comments reference `#create_objects`, but I don't see any overrides for that method anymore. Does that mean these comments can be removed? Or do they need to be updated to refer to `#create_collections` now?

summer-cook replied:
@bkiahstroud yes - just removed those comments!

summer-cook committed:
fix comment
